web/instances/fixers.go

```diff
@@ -381,27 +382,37 @@ func serviceTriggersFixer(c echo.Context) error {
 						continue
 					}
 					service.TriggerID = infos.TID
-					update = true
+					updateApp = true
 					break
 				}
 				recreate = service.TriggerID == ""
 			} else {
-				_, err := jobsSystem.GetTrigger(inst, service.TriggerID)
+				trigger, err := jobsSystem.GetTrigger(inst, service.TriggerID)
 				recreate = errors.Is(err, job.ErrNotFoundTrigger)
+				if err == nil {
+					var msg serviceMessage
+					if err := json.Unmarshal(trigger.Infos().Message, &msg); err != nil {
+						return err
+					}
+					if msg.Name == "" {
+						fixTriggerName(inst, trigger, msg, name)
+						updated++
+					}
+				}
 			}
 
 			if recreate {
-				triggerID, err := app.CreateServiceTrigger(inst, slug, service)
+				triggerID, err := app.CreateServiceTrigger(inst, slug, name, service)
```

Comment:
I don't understand why the service does not have a `name` here while the struct shows it should

Reply:
`name` is not exported as it starts with a lowercase letter, and as such, it is not filled when we unmarshal JSON. I was tricked too by this!